zebra/zebra_rnh.c

```diff
@@ -597,8 +601,8 @@ static int check_overlay_nexthop(struct prefix *pp, uint8_t *isreachable)
 	}
 
 	//get the nextop to check if it is connected. 
-	inet_ntop(pp->family, &pp->u.prefix, via, PREFIX2STR_BUFFER);
-	snprintf(cntrname, 256, "nh.%s", via);
+	inet_ntop(pp->family, &pp->u.prefix, nh, PREFIX2STR_BUFFER);
+	snprintf(cntrname, 256, "nh.%s", nh);
```

Comment:
:exclamation:**Cycode**: SAST violation: 'The software uses a signed primitive and performs a cast to an unsigned primitive, or uses an unsigned primitive and performs a cast to a signed primitive, which can produce an unexpected value. When the result of a function is to be used as a size parameter, using negative return values can have unexpected results. Although less frequent an issue, unsigned-to-signed conversion can be  the precursor to buffer underwrite conditions. Buffer underwrites  occur frequently when large unsigned values are cast to signed values,  and then used as indexes into a buffer or for pointer arithmetic.'.
 
Severity: Medium
 <!-- 9b776754-037c-4599-aecb-4e5c4ec771b5 -->